### PR TITLE
Feature/charts and dashboard test

### PR DESCRIPTION
**Pull request by siddhant3030**
1. Cleaned up dashboard and charts code
2. Wrote test cases for both charts and dashboard at schema, api and service layer
3. Removed raw and aggregated logic from everywhere 

Ideally this is how I'm thinking we should write tests

Schema Tests
Tests data validation at the input/output boundary.
1. Required fields are enforced
2. Field types are correct (string, int, list)
3. Nested structures validate properly
4. Default values work
5. Serialization (.dict()) works

Example: Does ChartCreate reject a request missing title?

Service Tests
Tests business logic in isolation from HTTP.
1. Permission checks (only creator can delete)
2. Complex operations (bulk delete with partial failures)
3. Edge cases (empty lists, cross-org access)
4. Custom exceptions are raised correctly
5. Database interactions work correctly

Example: Does ChartService.delete_chart() raise ChartPermissionError when a non-creator tries to delete?
API Tests

Tests the full request → response cycle.
1. HTTP status codes (200, 404, 400)
2. Authentication/authorization decorators
3. Request parsing and response formatting
4. Error messages returned to client

Example: Does GET /charts/99999/ return HTTP 404 with the right error message?

## Summary by CodeRabbit

* **New Features**
  * Centralized chart and dashboard service endpoints; dynamic map layers and map overlay access

* **Improvements**
  * Removed deprecated computation_type from chart payloads/responses; unified metrics-driven chart behavior and validation
  * Dashboard schemas and locking/share/landing-page flows standardized; clearer error responses and permission handling

* **Tests**
  * Extensive unit and end-to-end tests for charts, dashboards, schemas, services, and map flows

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: dev-secrets/warehouseCreds-my-company-e24fd386-326d-4425-8f2c-b1b388b7fcce [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "Name": "warehouseCreds-my-company-e24fd386-326d-4425-8f2c-b1b388b7fcce",
+  "SecretId": "warehouseCreds-my-company-e24fd386-326d-4425-8f2c-b1b388b7fcce",
+  "SecretString": "{\"host\": \"dalgo-staging-warehouses.c4hvhyuxrcet.ap-south-1.rds.amazonaws.com\", \"port\": 5432, \"database\": \"postgres\", \"username\": \"postgres\", \"password\": \"**********\", \"ssl\": false, \"ssl_mode\": {\"mode\": \"disable\"}, \"tunnel_method\": {\"tunnel_method\": \"NO_TUNNEL\"}}"
+}
```
[the remark this answers is not in the document]
Comment (siddhant3030) [reply dev-secrets/warehouseCreds-my-company-e24fd386-326d-4425-8f2c-b1b388b7fcce]: removed this

File: ddpui/services/dashboard_service.py [hunk 2]
```diff
@@ -30,9 +39,575 @@
...
+
+        logger.info(f"Created dashboard {dashboard.id} for org {orguser.org.id}")
+        return dashboard
+
+    @staticmethod
+    def update_dashboard(
+        dashboard_id: int,
+        org: Org,
+        orguser: OrgUser,
+        title: Optional[str] = None,
```
Comment (Ishankoradia) [inline ddpui/services/dashboard_service.py]: any reason why we aren't using the schema `DashboardUpdate` as an input in `update_dashboard` ?

same for other apis. we dont want to type the same payload as arguments if its already available.

thoughts ?

File: ddpui/services/chart_service.py [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+    def get_org_warehouse(org: Org) -> Optional[OrgWarehouse]:
+        """Get the warehouse for an organization.
+
+        Args:
+            org: The organization
+
+        Returns:
+            OrgWarehouse instance or None if not configured
+        """
+        return OrgWarehouse.objects.filter(org=org).first()
```
[the remark this answers is not in the document]
Comment (siddhant3030) [reply ddpui/services/chart_service.py]: we can def move this to warehouse service.

File: ddpui/core/charts/chart_validator.py [hunk 4]
```diff
@@ -133,222 +99,78 @@ def _validate_aggregate_function(aggregate_func: Optional[str]) -> None:
...
 
     @staticmethod
-    def _validate_bar_chart(
-        computation_type: str,
-        x_axis: Optional[str],
-        y_axis: Optional[str],
-        dimension_col: Optional[str],
-        metrics: Optional[List] = None,
+    def _validate_metrics(
+        metrics: Optional[List], chart_type: str, allow_multiple: bool = True
```
[the remark this answers is not in the document]
Comment (siddhant3030) [reply ddpui/core/charts/chart_validator.py]: done. you're right

File: ddpui/api/dashboard_native_api.py [hunk 5]
```diff
@@ -253,11 +154,9 @@ def get_dashboard(request, dashboard_id: int):
...
 
     try:
-        dashboard = Dashboard.objects.prefetch_related(
-            "filters", Prefetch("lock", queryset=DashboardLock.objects.all())
-        ).get(id=dashboard_id, org=orguser.org)
-    except Dashboard.DoesNotExist:
-        raise HttpError(404, "Dashboard not found")
+        dashboard = DashboardService.get_dashboard(dashboard_id, orguser.org)
+    except DashboardNotFoundError:
+        raise HttpError(404, "Dashboard not found") from None
```
[the remark this answers is not in the document]
Comment (siddhant3030) [reply ddpui/api/dashboard_native_api.py]: done.
